src/utils.js

```diff
@@ -247,9 +247,9 @@ export function evalSetCurrentStory({ waitFor }, story) {
     // resolve when rendered, reject on any other renderer event
     return new Promise((resolve, reject) => {
       channel.on('storyRendered', resolve);
-      channel.on('storyMissing', reject);
-      channel.on('storyErrored', reject);
-      channel.on('storyThrewException', reject);
+      channel.on('storyMissing', () => reject(new Error('Story Missing')));
```

Comment:
Do we get original error passed to us

Reply:
Nope, there are no args passed to this event.
I tried -
```js
channel.on('storyMissing', (...args) => console.log(args))
```
But this didn't log anything.

Reply:
Even tried -
```js
      channel.on('storyMissing', (err) => {
        reject(err);
      });
```
But this again resulted in -
`[percy] TypeError: Cannot read properties of undefined (reading 'message')`

Reply:
Even tried -
```js
      channel.on('storyMissing', (err) => {
        reject(err);
      });
```
But this again resulted in -
`[percy] TypeError: Cannot read properties of undefined (reading 'message')`

Reply:
Oh, just learned something new about JS which explains this better.
Whenever a function name is passed as callback, it calls the function by whatever parameters are passed.
So, in this case
```js
channel.on('storyMissing', reject)
```

This would call reject with whatever parameters are passed to eventListener, and since nothing is passed, hence we reject with `undefined` resulting in above error.

Reply:
> so basically story missing isnt passing anything ... 

yes

> Also its not a new thing, its a standard thing that works across almost all languages ( except where brackets are not required for a function call, Ruby 👀 ), it works because its no different, it expects a function and you are passing a function

Wasn't aware of this, thanks!